Server/ReasnAPI/ReasnAPI/Services/ParticipantService.cs

```diff
@@ -1,16 +1,98 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class ParticipantService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using ReasnAPI.Services.Exceptions;
+using System.Linq.Expressions;
+
+namespace ReasnAPI.Services;
+
+public class ParticipantService(ReasnContext context)
+{
+    public ParticipantDto? CreateParticipant(ParticipantDto? participantDto)
+    {
+        ArgumentNullException.ThrowIfNull(participantDto);
+
+        var participantDb = context.Participants.FirstOrDefault(r => r.Event.Id == participantDto.EventId && r.User.Id == participantDto.UserId);
+
+        if (participantDb is not null)
+        {
+            throw new BadRequestException("Participant already exists");
+        }
+
+        var userInDb = context.Users.FirstOrDefault(r => r.Id == participantDto.UserId);
+        var eventInDb = context.Events.FirstOrDefault(r => r.Id == participantDto.EventId);
+
+        if (userInDb is null)
+        {
+            throw new NotFoundException("User not found");
+        }
+
+        if (eventInDb is null)
+        {
+            throw new NotFoundException("Event not found");
+        }

```

Comment:
Same as above, either do one common query and one check, or: query -> check -> query -> check.